### DISCUSSION TITLE
ExcelDna quick start

### DISCUSSION THREAD

**Discussion by kuankuan27** [Excel-DNA/ExcelDna#678] [Q&A] [answered]
Hi,

I'm really new to C# and ExcelDna, and I'm excited to come across this tool. I'm trying to follow the start-up guide: https://excel-dna.net/docs/getting-started I have some very basic questions.

I guess since the start-up guide didn't mention which Nuget package to get, I got the ExcelDna.Integration 1.7.0. I was met with the warnings that it was restored using .NetFramework, Version=v.4.8.1 etc. I guess I can ignore this warning?

I'm using the Visual Studio 2022, and I couldn't find debug with excel. Instead I debugged by attaching to an existing running Excel process, but nothing happens to my excel. 

Should I use an earlier version of Visual Studio?

**Comment by govert** [comment, marked as answer]
The main package to reference is called `ExcelDna.AddIn`, and version 1.7.0 is the current release. Visual Studio 2022 is best. It's better these days to make an SDK-style project file, which you get by picking the "Class Library" project template instead of the "Class Library (.NET Framework)" template. After the project is create, just change the `TargetFramework` to "net472" or "net48" to target .NET Framework, which is simplest. Then your NuGet package references go into the project files as `<PackageReference>` items and you don't have a packages.config file. Having done all of this, you can press F5 to build and debug (sometimes need to build twice to get rid of a VS bug after you change the TargetFramework). You can also attach to a running Excel (make sure you pick the right debugging engine) and the File -> Open the right .xll file in the output directory (for 32-bit or 64-bit).

**Reply by kuankuan27** [reply]
Thank you for the quick response! and I had it working. However, I see that after opening a new library file, there is no selection for net48 for target framework (even if I click the small link under the selections for manual installation). I had to open a new project that says library (.net framework 4.8) to have ExcelDna.addin working. I hope this still fine.

**Reply by govert** [reply]
The "Class Library" template (without the extra "(.NET Framework)" part in the name, gives you an SDK-style project file. You can't select .NET Framework 4.8 when creating it, but you can edit the project file afterwards. A simple project file would then look like this:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net48</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="ExcelDna.AddIn" Version="1.7.0" />
  </ItemGroup>

</Project>
```
